pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -76,7 +77,7 @@ func fixBundleExtracted() error {
 	if err := os.Chmod(constants.MachineCacheDir, 0775); err != nil {
 		logging.Debugf("Error changing %s permissions to 0775", constants.MachineCacheDir)
 	}
-	if constants.BundleEmbedded() {
+	if version.IsRelease() {
```

Comment:
I think this line will add an extra copy for the pkg. 
The crcbundle is already in /Library and will be now copied in ~/.crc/cache before extraction.

Reply:
We can only start extracting the bundle if it is embedded so may be with release, we should also use embedded logic?
```
if version.IsRelease() && constants.BundleEmbedded{
}
```

Reply:
This indeed looks like this needs more work... I think something like this is more correct though:
```
if version.IsRelease() {
		bundleDir := filepath.Dir(constants.DefaultBundlePath)
		if err := os.MkdirAll(bundleDir, 0775); err != nil {
			return fmt.Errorf("Cannot create directory %s: %v", bundleDir, err)
		}
		if constants.BundleEmbedded {
			if err := embed.Extract(filepath.Base(constants.DefaultBundlePath), constants.DefaultBundlePath); err != nil {
				return err
			}
		}
		_, err := bundle.Extract(constants.DefaultBundlePath)
		return err
	}
```

But this makes me wonder why we have this code in the first place https://github.com/code-ready/crc/blob/de1d9f11eea9c10e822aa2dbbf040aaf722ae6de/pkg/embed/embed.go#L16-L21 if we never want to do an extra copy of the bundle.

Reply:
Yeah.. the copy of small binaries was less noticeable for the end user. It should be better to override binDir totally.

Reply:
Actually with the current code, `fixBundleExtracted` will never be called in the macos package case. 
```
func checkBundleExtracted() error {
	if !version.IsRelease() {
		return nil
	}
	if _, err := os.Stat(constants.DefaultBundlePath); os.IsNotExist(err) {
		return err
	}
	return nil
}
```

`constants.DefaultBundlePath` points at `/Library/CodeReady  Containers/....` which will always exist after package install.